### PR TITLE
chore: Add `cargo-hack` to check features

### PR DESCRIPTION
**Pull request by LucioFranco**
r? @taiki-e this crate is awesome :) 

Signed-off-by: Lucio Franco <luciofranco14@gmail.com>

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -24,18 +24,14 @@ jobs:
...
-    - name: Check with transport no tls
-      run: cargo check -p tonic
-    - name: Check with transport w/ openssl
-      run: cargo check -p tonic --features openssl
-    - name: Check with transport w/ rustls
-      run: cargo check -p tonic --features rustls
+    - name: Check features
+      run: cargo hack check --all --each-feature
+    - name: Check all targets
+      run: cargo check --all --all-targets --all-features
```
Comment (LucioFranco) [inline .github/workflows/CI.yml]: I need to add the `all-targets` or else it will miss benchmarks, do you think there might be a better solution @taiki-e?

Comment (taiki-e) [reply .github/workflows/CI.yml]: Hmm, I'm not sure if there is another good solution. (afaik, `--all-targets` is basically [the same as `--lib --bins --examples --tests --benches`](https://github.com/rust-lang/cargo/blob/6a7f505a185b000e38bdad64392098e0b2e50802/src/cargo/ops/cargo_compile.rs#L551-L560), and note that the document is not checked (https://github.com/rust-lang/cargo/issues/6669)

Comment (LucioFranco) [reply .github/workflows/CI.yml]: if we run it last it should be fine I can wait till we fine a better solution.

**Comment by LucioFranco** [comment]
@taiki-e thank you so much for this crate, its what I very much wanted but had no clue how to express what I wanted. <3

**Pull request merged**